eth/beacon/state_machines/forks/serenity/operations.py

```diff
@@ -2,12 +2,51 @@
 from eth.beacon.types.pending_attestation_records import PendingAttestationRecord
 from eth.beacon.types.states import BeaconState
 from eth.beacon.state_machines.configs import BeaconConfig
+from eth.beacon.helpers import get_beacon_proposer_index
+from eth._utils.numeric import bitwise_xor
 
 from .validation import (
     validate_serenity_attestation,
+    validate_serenity_randao_reveal,
 )
 
 
+def validate_randao(state: BeaconState,
```

Comment:
1. Move this function to new file `eth/beacon/state_machines/forks/serenity/block_transtions.py`.
2. Rename it to `process_randao`.